src/data/config-codec.ts

```diff
@@ -7,7 +9,36 @@ export const configCodec = oneOf([
         administratorGroups: array(string),
         ouLabel: string,
         periodLabel: string,
-        infoPlaceholder: optional(string),
+        messageInfo: oneOf([optional(string), record(string, string)]),
     }),
     Codec.interface({ administratorGroups: array(string) }), // Backwards compatibility
 ]);
+
+export function decodeConfig(json: unknown, storage: string, key: string): Config {
+    return configCodec.decode(json).caseOf({
+        Left: (err): Config => {
+            const errStr = errors.invalidJSON(storage, key);
+            console.error(new Error(errStr + err));
+            throw new Error(errStr);
+        },
+        Right: (res): Config => {
+            if ("sheetName" in res) {
```

Comment:
Ugly processing, but I guess we have to support legacy configs.

Reply:
I agree... perhaps separating the conditions into meaningful parts could look a bit better? @tokland